### PR TITLE
ci: Support using git for dependencies in `test-installation`

### PR DESCRIPTION
**Pull request by Marenz**
Fix the CI test that fails when dependencies in pyproject.toml contain git urls.

I didn't have time to update the tests, feel free to take over.

### CODE REVIEW

**Comment by llucax** [comment]
Updating the tests should be just: `UPDATE_GOLDEN=1 pytest tests/integration/test_cookiecutter_generation.py::test_golden`.

Ideally this repo should get the fix.

I plan to dedicate some time to repo-config in a not so distant future, if you didn't take care of it until then I can take it over.

**Review by llucax — CHANGES_REQUESTED** [review]
Marking as needing changes so it is more obvious in the PR list. Basically this needs:

- Update this project with the fix
- Fix tests
- Split the cleaning of the docker image in its own commit
- Improve the commit description a bit

**Review by llucax — DISMISSED** [review]

**Comment by llucax** [comment]
OK, I took over this one and updated with the missing stuff. But now it needs an approval from someone else :)

**Comment by Marenz** [comment]
Yeah, I can't approve this, being the author and all.

**Review by cwasicki — APPROVED** [review]

**Review by leandro-lucarella-frequenz — APPROVED** [review]

**Review by shsms — APPROVED** [review]

**Pull request merged**
